eppo_core/src/event_ingestion/delivery.rs

```diff
@@ -80,12 +80,13 @@ pub(super) async fn delivery(
                 match err {
                     EventDeliveryError::RetriableError(_) => {
                         // Retry later
-                        deliver_status(&delivery_status, QueuedBatch::failure(batch)).await;
+                        deliver_status(&delivery_status, QueuedBatch::retry(batch)).await;
                     }
-                    EventDeliveryError::NonRetriableError(_) => {
+                    _ => {
```

Comment:
still need to do more fine grained error handling here, will address in future PR

Reply:
In the case of `EventPayloadTooLargeError`, we could potentially do either:
1. Just throw away just the event(s) that failed and retry the rest of the batch
2. Instead of returning `EventPayloadTooLargeError`, we can instead have `EventDelivery` log an error and discard those offending events but still deliver the rest of the batch. The downside of this approach is that it 

I'm kinda leaning towards option 2 here but the downside is that it doesn't give the user the opportunity to fix it and it may go unnoticed until they check the application logs. thoughts @rasendubi @petzel ?

Reply:
ended up doing something different altogether

Reply:
ended up doing something different altogether